src/main/java/edu/rosehulman/covidtracer/service/EmailService.java

```diff
@@ -0,0 +1,120 @@
+import com.sendgrid.helpers.mail.objects.Personalization;
+import edu.rosehulman.covidtracer.model.Person;
+import edu.rosehulman.covidtracer.repository.PersonRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+@Service
+public class EmailService {
+
+    public static final String SERVICE_EMAIL = "rhitcovidtracer@gmail.com";
+
+    @Autowired
+    PersonService personService;
+
+    @Value("${sendgrid.api-key}")
+    private String apiKey;
+
+    @Value("${email.template.basic}")
+    public String basicTemplate;
+
+    public Response sendEmail(Person recipient, String subject, String templateId) throws IOException {
+        Email fromAddress = new Email(SERVICE_EMAIL);
+        Email toAddress = new Email(recipient.getEmail());
+        Content content = new Content("text/html", "Covid Tracer Official Email");
+        Mail mail = new Mail(fromAddress, subject, toAddress, content);
+        mail.setTemplateId(templateId);
+        Personalization p = new Personalization();
+        String toName = recipient.getFullName();
+        p.addDynamicTemplateData("first_name", toName);
+        p.addTo(toAddress);
+        mail.personalization.add(p);
+
+        SendGrid sg = new SendGrid(apiKey);
+        Request request = new Request();
+        try {
+            request.setMethod(Method.POST);
+            request.setEndpoint("mail/send");
+            request.setBody(mail.build());
+            Response r = sg.api(request);
+            System.out.println("Response: " + r.getStatusCode() + " " + r.getBody());
+            return r;
+        }
+        catch (IOException ex) {
+            throw ex;
+        }
+
+    }
+
+    public String getBasicTemplate(){
+        return this.basicTemplate;
+    }
+
+
+    public Response sendEmail(String from, String subject, String to, String templateId) throws IOException{
+        Email fromAddress = new Email(from);
+        Email toAddress = new Email(to);
+        Content content = new Content("text/html", "Covid Tracer Stuff");
+        Mail mail = new Mail(fromAddress, subject, toAddress, content);
+        mail.setTemplateId(templateId);
+        Personalization p = new Personalization();
+        String toName = personService.getNameFromEmail(to);
+        p.addDynamicTemplateData("first_name", toName);
+        p.addTo(toAddress);
+        mail.personalization.add(p);
+
+        SendGrid sg = new SendGrid(apiKey);
+        Request request = new Request();
+        try {
+            request.setMethod(Method.POST);
+            request.setEndpoint("mail/send");
+            request.setBody(mail.build());
+            Response r = sg.api(request);
+            System.out.println("Response: " + r.getStatusCode() + " " + r.getBody());
+            return r;
+        }
+        catch (IOException ex) {
+            throw ex;
```

Comment:
Re: catch and throw

This feel like that shortstop that the outfielder throws too, I think they call them the cut off.